pkgs/stdenv/generic/setup.sh

```diff
@@ -12,7 +12,11 @@ fi
 
 shopt -s inherit_errexit
 
-if (( "${NIX_DEBUG:-0}" >= 6 )); then
+# $NIX_DEBUG must be a documented integer level, if set, so we can use it safely as an integer.
+# See the `Verbosity` enum in the Nix source for these levels.
+[[ -z ${NIX_DEBUG-} || $NIX_DEBUG == [0-7] ]]
```

Comment:
Can we `echo` and `exit` in this case (similar to Bash version check above)? Otherwise `setup.sh` would fail without any message.

Reply:
Yeah, good call -- I'll do that.